lib/restAPIs/library.js

```diff
@@ -0,0 +1,108 @@
+const assert = require('assert');
+
+const assertParam = require('../util/assertParam');
+const RestUtil = require('../util/RestUtil');
+const buildCommonQuery = require('../util/buildCommonQueryParams');
+
+module.exports = class RestTopic extends RestUtil {
+  /**
+   * Get base endpoint for this API category
+   */
+  _getBase() {
+    return 'library';
+  }
+
+  /**
+   * Return object representing the library for the authenticated user.
+   * @param {String} outputFlag - Filtered output. DEAFULT = include everything, FILTER_TOC = filter out cahpters and pages.
+   * @param {String} fields - Fields to be included in the result
+   * @return {Object} Library for the authenticated user
+   */
+  getLibrary(outputFlag = 'DEFAULT', fields = '') {
+    const queryParameter = buildCommonQuery(fields);
+    queryParameter.outputFlag = outputFlag;
+
+    return this._makeRequest(
+      this._getBase(),
+      queryParameter,
+      'GET'
+    ).then(result => this.throwIfFailed(result, 200));
+  }
+
+  /**
+   * Publish an object (document or dossier) in project defined by projectID.
+   * Information about object to publish and recipient are detailed in object 'body'.
+   * https://demo.microstrategy.com/MicroStrategyLibrary/api-docs/index.html#/Library/publishObject
+   * @param {String} projectID -
+   * @param {object} body
```

Comment:
Param order is reversed (body first then projectID). Also, `object` -> `Object` and `String` -> `string`

Reply:
Done.